### PR TITLE
Fix bugs in sod shocktube.

### PR DESCRIPTION
**Pull request by abhinavmuta**
- Before both solids and mirror bc are set which is incorrect.
 - Solid support is removed as mirror bc is sufficient for this problem.

### CODE REVIEW

File: pysph/examples/gas_dynamics/sod_shocktube.py [hunk 1]
```diff
@@ -50,41 +49,52 @@ def consume_user_options(self):
...
-            beta=2.0, update_alpha1=True, update_alpha2=True
+            fluids=['fluid'], solids=[], dim=dim, gamma=gamma,
+            kernel_factor=None, alpha1=1.0, alpha2=0.1,
+            beta=2.0, update_alpha1=True, update_alpha2=True,
         )
         gsph = GSPHScheme(
-            fluids=['fluid'], solids=['boundary'], dim=dim, gamma=gamma,
-            kernel_factor=1.0,
+            fluids=['fluid'], solids=[], dim=dim, gamma=gamma,
+            kernel_factor=None,
```
Comment (psnbaba) [inline pysph/examples/gas_dynamics/sod_shocktube.py:90]: Setting this to None will effect values of UpdateSmoothingLengthFromVolume equation.

File: pysph/examples/gas_dynamics/sod_shocktube.py [hunk 2]
```diff
@@ -50,41 +49,52 @@ def consume_user_options(self):
...
-            kernel_factor=1.2, alpha1=1.0, alpha2=0.1,
-            beta=2.0, update_alpha1=True, update_alpha2=True
+            fluids=['fluid'], solids=[], dim=dim, gamma=gamma,
+            kernel_factor=None, alpha1=1.0, alpha2=0.1,
+            beta=2.0, update_alpha1=True, update_alpha2=True,
         )
         gsph = GSPHScheme(
-            fluids=['fluid'], solids=['boundary'], dim=dim, gamma=gamma,
-            kernel_factor=1.0,
+            fluids=['fluid'], solids=[], dim=dim, gamma=gamma,
```
Comment (psnbaba) [inline pysph/examples/gas_dynamics/sod_shocktube.py:89]: setting `solids=[]` makes empty groups for example see https://github.com/pypr/pysph/blob/7e0b5d5eb8a925943623210799a9cec436e142d9/pysph/sph/scheme.py#L1323

File: pysph/examples/gas_dynamics/sod_shocktube.py (hunk 1 above)
Comment (abhinavmuta) [reply pysph/examples/gas_dynamics/sod_shocktube.py:90]: This value should be same as the smoothing length factor hdx. I am setting it to hdx in configure_scheme.

File: pysph/examples/gas_dynamics/sod_shocktube.py (hunk 2 above)
Comment (abhinavmuta) [reply pysph/examples/gas_dynamics/sod_shocktube.py:89]: Thanks. I've put a condition here.

**Comment by prabhuramachandran** [comment]
LGTM, thanks!

**Pull request merged**